operator/pkg/controlplane/apiserver/apiserver.go

```diff
@@ -17,23 +17,23 @@ import (
 )
 
 // EnsureKarmadaAPIServer creates karmada apiserver deployment and service resource
-func EnsureKarmadaAPIServer(client clientset.Interface, cfg *operatorv1alpha1.KarmadaComponents, name, namespace string) error {
-	if err := installKarmadaAPIServer(client, cfg.KarmadaAPIServer, name, namespace); err != nil {
+func EnsureKarmadaAPIServer(client clientset.Interface, cfg *operatorv1alpha1.KarmadaComponents, name, namespace string, featureGates map[string]bool) error {
```

Comment:
I wonder if there will be other extension options in addition to featureGates.
If yes, I'd like to encapsulate it into a struct,

Reply:
This a good point. 

Currently, It seem not have extra arguments. If there are more options, we can optimize to a struct in the future.